### PR TITLE
Feature/each date once

### PR DESCRIPTION
**Pull request by Mark-Krensel**
Hello, please review my changed data structure.

### CODE REVIEW

File: components/Card.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ export default function Card({
   feastTime,
 }) {
   const { pathname } = useRouter();
-
+  weight.forEach((element) => console.log(element));
```
Comment (F-Kirchhoff) [inline components/Card.js]: 🕵️

File: components/Card.js [hunk 2]
```diff
@@ -21,13 +21,29 @@ export default function Card({
...
       )}
       <p>Date: {date}</p>
-      {weight && <p>Weight: {weight}</p>}
-      {height && <p>Height: {height}</p>}
-      {feastTime && (
-        <p>
-          Nurse time: {feastTime.substr(0, 2)}:{feastTime.substr(2, 2)}:
-          {feastTime.substr(4, 2)}
-        </p>
+      {weight?.[0] && (
```
Comment (F-Kirchhoff) [inline components/Card.js]: the naming of your "weight" variable is confusing. Are that multiple items? If so please rename it to weightList or weights.

Comment (F-Kirchhoff) [inline components/Card.js]: You don't need the optional chaining operator here. if the first entry is not available if will default to `undefined`. The best method here would be weightList.length === 0 && ...

File: components/Card.js [hunk 3]
```diff
@@ -21,13 +21,29 @@ export default function Card({
...
+        <ul>
+          {height.map((height, index) => (
+            <li key={index}>Height: {height} </li>
+          ))}
+        </ul>
+      )}
+      {feastTime?.[0] && (
+        <ul>
+          {feastTime.map((entry, index) => (
+            <li key={index}>
```
Comment (F-Kirchhoff) [inline components/Card.js]: please don't use the index here, get an index from your database for that. (Tipp: to display you don't have to chop your datapoints into weight, height, etc... you can hand over the whole object with all information and then only use the parts that are insteresting for that specific list).

File: components/Card.js [hunk 4]
```diff
@@ -21,13 +21,29 @@ export default function Card({
...
+      {height?.[0] && (
+        <ul>
+          {height.map((height, index) => (
+            <li key={index}>Height: {height} </li>
+          ))}
+        </ul>
+      )}
+      {feastTime?.[0] && (
+        <ul>
+          {feastTime.map((entry, index) => (
```
Comment (F-Kirchhoff) [inline components/Card.js]: feast means something like eating at a big buffet or restaurant, I guess you mean feedingTime here?

File: components/Card.js [hunk 5]
```diff
@@ -21,13 +21,29 @@ export default function Card({
...
+          ))}
+        </ul>
+      )}
+      {height?.[0] && (
+        <ul>
+          {height.map((height, index) => (
+            <li key={index}>Height: {height} </li>
+          ))}
+        </ul>
+      )}
```
Comment (F-Kirchhoff) [inline components/Card.js]: did you know that if your array is empty you can still "map" over it? I won't produce an outcome of course, but not an error either. So you don't even have to check if the array has some values, just map over all items, if the array happens to be empty no list items will be displayed.

File: pages/api/Days/index.js [hunk 6]
```diff
@@ -13,17 +13,61 @@ export default async function handler(request, response) {
...
+          updatedDay
+        );
+
+        return response
+          .status(201)
+          .json({ message: "Data saved", updatedId: updatedDayInDb.id });
+      } else {
+        postData.weight = postData.weight ? [postData.weight] : [];
+        postData.height = postData.height ? [postData.height] : [];
+        postData.feastTime = postData.feastTime ? [postData.feastTime] : [];
```
Comment (F-Kirchhoff) [inline pages/api/Days/index.js]: please don't mutate your postData here, create a new variable called "newDayData"

File: services/dayService.js [hunk 7]
```diff
@@ -34,3 +34,19 @@ export async function getDayById(id) {
...
+  if (day) {
+    const sanitizedDay = {
+      id: day.id,
+      date: day.date,
+      weight: day.weight,
+      height: day.height,
+      feastTime: day.feastTime,
+    };
+    return sanitizedDay;
+  } else console.log("day not in db");
```
Comment (F-Kirchhoff) [inline services/dayService.js]: since you return in the if statement, you can remove the "else" here.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Mark, very nice stuff here! I only found some minor issues, nothing big. Have a look!

File: components/Card.js (hunk 1 above)
Comment (Mark-Krensel) [reply components/Card.js]: done

File: components/Card.js (hunk 2 above)
Comment (Mark-Krensel) [reply components/Card.js]: done

Comment (Mark-Krensel) [reply components/Card.js]: thanks for the hint, I changed it to ```array.lenght !== 0```  and implemented it

File: components/Card.js (hunk 4 above)
Comment (Mark-Krensel) [reply components/Card.js]: I guess this was just a poor joke from my side :(

File: components/Card.js (hunk 5 above)
Comment (Mark-Krensel) [reply components/Card.js]: I definitely  didn't know that. But then the ```<ul>``` will still show up. Is it ok to put everything into one big ```<ul>``` ?

File: components/Card.js (hunk 3 above)
Comment (Mark-Krensel) [reply components/Card.js]: I'm not sure I understand your feedback on this one. I thought I am already getting the index from that array, which comes from my db?

File: components/Card.js (hunk 2 above)
Comment (Mark-Krensel) [reply components/Card.js]: It seems I really do need the optional chaining operator here, as on the attributes pages (weights for now) i dont pass all of the arrays down

Comment (Mark-Krensel) [reply components/Card.js]: ok, now I'm handing it an empty array

File: components/Card.js (hunk 5 above)
Comment (Mark-Krensel) [reply components/Card.js]: I want to give each ul a header, so I think I need to keep it as is

File: models/Day.js [hunk 8]
```diff
@@ -1,16 +1,38 @@
...
+        id: { type: mongoose.Types.ObjectId },
+      },
+    ],
+    feastTimes: [
+      {
+        timeStamp: { type: String },
+        value: { type: String },
+        id: { type: mongoose.Types.ObjectId },
+      },
+    ],
```
Comment (F-Kirchhoff) [inline models/Day.js:35]: nice! if you specify this type ObjectId, does mongoose generate the id automatically for you?

File: pages/weight.js [hunk 9]
```diff
@@ -13,34 +13,44 @@ export async function getServerSideProps() {
...
-  const chartData = ascendingFilteredDays.map((day) => day.weight);
+  const chartData = ascendingFilteredDays.map((day) =>
+    day.weights.map((weight) => weight.value)
+  );
+
   const meanChartData = chartData.map(
     (array) => array.reduce((a, b) => a + b, 0) / array.length
   );
-  const mergedChartData = meanChartData.flat(1);
+  console.log(meanChartData);
```
Comment (F-Kirchhoff) [inline pages/weight.js]: 🕵️

File: pages/weight.js [hunk 10]
```diff
@@ -13,34 +13,44 @@ export async function getServerSideProps() {
...
-        {filteredDays.map((filteredDay) => (
-          <Card
-            key={filteredDay.id}
-            date={filteredDay.date}
-            weight={filteredDay.weight}
-          />
-        ))}
+        {filteredDays.map(
+          (filteredDay) => (
+            console.log(filteredDay),
```
Comment (F-Kirchhoff) [inline pages/weight.js]: 🕵️

File: services/dayService.js [hunk 11]
```diff
@@ -43,10 +43,11 @@ export async function checkAndGetDate(dateToBeChecked) {
...
-      height: day.height,
-      feastTime: day.feastTime,
+      weights: day.weights,
+      heights: day.heights,
+      feastTimes: day.feastTimes,
     };
     return sanitizedDay;
-  } else console.log("day not in db");
+  }
+  console.log("day not in db");
```
Comment (F-Kirchhoff) [inline services/dayService.js]: you should use a console.error here or remove the log

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Only found some console.logs, otherwise its ready to go!

File: models/Day.js (hunk 8 above)
Comment (Mark-Krensel) [reply models/Day.js:35]: yes it does, but you need to call with ```object._id``` , apparently it doesn't work with just ```object.id```

File: pages/weight.js (hunk 9 above)
Comment (Mark-Krensel) [reply pages/weight.js]: done

File: pages/weight.js (hunk 10 above)
Comment (Mark-Krensel) [reply pages/weight.js]: done

File: services/dayService.js (hunk 11 above)
Comment (Mark-Krensel) [reply services/dayService.js]: done

**Review by F-Kirchhoff — APPROVED** [review]
LGTM 🦊

**Pull request merged**